### PR TITLE
adding constant for annotation

### LINKED ISSUE #1407: Add constants for supported duck labels and annotations
**Problem**
We don't have any constant set for common labels and annotations such as
* duck.knative.dev/addressable
* duck.knative.dev/source
* ...

This can lead to repetition in the code base, especially in projects that use Knative as a dependency.

**[Persona:](https://github.com/knative/eventing/blob/master/docs/personas.md)** 
Contributors

**Exit Criteria**
Supported labels and annotations are defined as public constants inside `pkg/apis/duck`

Example: https://github.com/knative/serving/blob/3b5be63/pkg/apis/serving/register.go

**Time Estimate (optional):** 
How many developer-days do you think this may take to resolve?
1/2

### PR DESCRIPTION
**Pull request by matzew**
Signed-off-by: Matthias Wessendorf <mwessend@redhat.com>

fixes #1407

### CODE REVIEW

File: apis/duck/register.go [hunk 1]
```diff
@@ -18,4 +18,12 @@ package duck
 
 const (
 	GroupName = "duck.knative.dev"
+
+	// AddressableDuckVersionAnnotation is the annotation we use to declare
```
Comment (Abd4llA) [inline apis/duck/register.go:22]: I think it's a label?

Comment (antoineco) [reply apis/duck/register.go]: Agree, "Label" is a more term.

File: apis/duck/register.go [hunk 2]
```diff
@@ -18,4 +18,12 @@ package duck
 
 const (
 	GroupName = "duck.knative.dev"
+
+	// AddressableDuckVersionLabel is the label we use to declare
+	// which Addressable duck version type we conform to.
```
Comment (antoineco) [inline apis/duck/register.go:23]: It's only used with the value `true` (could even be an empty string) so it's not about the version AFAIK. Or did I miss an update?
```suggestion
	// that a type conforms to the Addressable duck type.
```

**Comment by matzew** [comment]
@mattmoor @vaikas mind taking a looksie here ? thanks!

File: apis/duck/register.go [hunk 3]
```diff
@@ -18,4 +18,12 @@ package duck
...
+	AddressableDuckVersionLabel = "duck.knative.dev/addressable"
+
+	// SourceDuckVersionLabel is the label we use to declare
+	// which Source duck version type we conform to.
```
Comment (antoineco) [inline apis/duck/register.go:27]: Same comment.
```suggestion
	// that a type conforms to the Source duck type.
```

**Pull request merged**